### PR TITLE
Stacked Promo variation for opera-mini

### LINKED ISSUE #9796: Horizontal scroll promos - not scrollable on Opera Mini Extreme
**Describe the bug**
Horizontal scroll does not work for multi-card promos.
Any cards beyond the first are not fully visible. Content is not accessible. 

**To Reproduce**
Steps to reproduce the behaviour:

1. Go to links on test:
* https://www.test.bbc.com/arabic/articles/cjn2e0jek55o (RTL script) 
* https://www.test.bbc.com/pidgin/articles/crrrkxz2k0ko
2. Scroll down to view multi-card promos
3. Attempt to scroll with horizontal finger swipe

**Expected behaviour**
Row of cards should be scrollable to show additional cards and allow links to be clicked. 

**Smartphone (please complete the following information):**

- Device: Google Pixel 5A
- OS: Android
- Browser: Opera Mini - Extreme Mode
- Version:  61.x

**Testing notes**

- [ ] This bug fix is expected to need manual testing.

**Checklist**
- [ ] (BBC contributors only) This issue follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Additional context**
See original issue/ feature request ticket #9697 
Fixed by Pull Request #9729

A Psammead issue [#3099](https://github.com/bbc/psammead/issues/3099) was previously raised, affecting scrollable banner menu.
This was not fixed as the hidden content can still be accessible via the burger menu.

### PR DESCRIPTION
**Pull request by DarioR01**
Resolves #9796 

**Overall change:**
Added stacked promo styling for opera mini browser as a workaround for the not working scrollbar

**Code changes:**

- Added flex-direction to be column in opera-mini
- Single card width to be 100% of the article with margin-right equal to the margin left in opera-mini
- Imported and used useOperaMiniDetection to apply correct style to opera-mini only.
- use useOperaMiniDetection hook to detect if the component is running under opera mini browser.

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project
- [ ] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by Louis-Matsika — APPROVED** [review]

**Review by ryanmccombe — APPROVED** [review]
👍 

Big conditionals in emotion strings can be hard to follow, so sometimes easier to just create two components, eg:

```javascript
const Component = isOperaMini ? OperaPromoBox : PromoBox;
```

Can also share styles by inheritance:

```javascript
const OperaPromoBox = styled(PromoBox)``
```

**Comment by DarioR01** [comment]
> 👍
> 
> Big conditionals in emotion strings can be hard to follow, so sometimes easier to just create two components, eg:
> 
> ```js
> const Component = isOperaMini ? OperaPromoBox : PromoBox;
> ```
> 
> Can also share styles by inheritance:
> 
> ```js
> const OperaPromoBox = styled(PromoBox)``
> ```

🙇 Thanks for providing me more options :D. I will do a refactor following your suggestions before I merge this.
